src/comm/ModDevPoll.cc

```diff
@@ -48,9 +48,8 @@
 
 #define DEBUG_DEVPOLL 0
 
-// OPEN_MAX is defined in <climits>
-#define DEVPOLL_UPDATESIZE  OPEN_MAX
-#define DEVPOLL_QUERYSIZE   OPEN_MAX
```

Comment:
What about SQUID_MAXFD at the very beginning of Comm::SelectLoopInit()? Should that become Squid_MaxFD as well? If not, it would be best to add a C++ comment there explaining why the three contexts (i.e. devpoll_state, do_poll, and devpoll_update) should use two different counters (i.e. Squid_MaxFD and SQUID_MAXFD).

Reply:
I'm new to squid code, so I don't quite understand the relationship between SQUID_MAXFD and Squid_MaxFD - I see that SQUID_MAXFD is set at compile time, and Squid_MaxFD is set at runtime, but I don't see what SQUID_MAXFD is used for outside of making sure that if USE_SELECT is defined, that FD_SETSIZE is at least SQUID_MAXFD.

Reply:
I have not checked all the details, but, IIRC, the following is true about Squid_MaxFD:

* Squid_MaxFD is the maximum number of open descriptors a running Squid process supports;
* Squid_MaxFD is the size of fde::Table (a.k.a. fd_table; see fde::Init());
* all open descriptors are smaller than Squid_MaxFD.

AFAICT, that is all we need to know to answer the question in this change request.